test-poc/framework/src/main/java/org/keycloak/test/framework/events/SysLogServer.java

```diff
@@ -0,0 +1,79 @@
+package org.keycloak.test.framework.events;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.ServerSocket;
+import java.net.Socket;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Set;
+
+public class SysLogServer {
+
+    private static final int MAX_THREADS = 5;
+    private ServerSocket serverSocket;
+    private final List<Thread> threads = Collections.synchronizedList(new LinkedList<>());
+    private final Set<SysLogListener> listeners = new HashSet<>();
+    private boolean running = true;
+
+    public void start() throws IOException {
+        serverSocket = new ServerSocket(0);
+        startThread();
+    }
+
+    public void stop() throws InterruptedException, IOException {
+        running = false;
+        serverSocket.close();
+        for (Thread t : threads) {
+            t.join();
+        }
+    }
+
+    public void addListener(SysLogListener listener) {
+        listeners.add(listener);
+    }
+
+    public void removeListener(SysLogListener listener) {
+        listeners.remove(listener);
+    }
+
+    public String getEndpoint() {
+        return "localhost:" + serverSocket.getLocalPort();
+    }
+
+    protected void startThread() {
+        if (running) {
+            Thread thread = new Thread(new BasicSocketHandler());
+            thread.start();
+            threads.add(thread);
+        }
+    }
+
+    private class BasicSocketHandler implements Runnable {
+
+        @Override
+        public void run() {
+            while (running) {
+                try {
+                    Socket socket = serverSocket.accept();
+                    if (threads.size() < MAX_THREADS) {
```

Comment:
Is it possible to use multiple sockets when no parallelism is enabled? Or it is by design to support JUnit5 parallel execution -> methods.